File: GameAssistant/web/backend/app/schemas/inference.py
"""Pydantic schemas for inference history."""
from datetime import datetime
from typing import Optional
from pydantic import ConfigDict

from app.schemas._base import NoProtectedNamespaceModel


class DetectionItemSchema(NoProtectedNamespaceModel):
    cls: str
    x: float
    y: float
    w: float
    h: float
    conf: float
    source: str


class InferenceResultBase(NoProtectedNamespaceModel):
    name: Optional[str] = None
    source_type: str = "device"
    source_file: Optional[str] = None
    model_id: Optional[str] = None
    inference_mode: str = "hybrid"
    confidence_threshold: float = 0.25


class InferenceResultCreate(InferenceResultBase):
    total_detections: int = 0
    detections_json: list[DetectionItemSchema] = []
    inference_time_ms: Optional[int] = None
    image_path: Optional[str] = None
    annotated_image_path: Optional[str] = None
    video_frame_index: Optional[int] = None
    device_id: Optional[str] = None


class InferenceResultResponse(InferenceResultBase):
    id: str
    total_detections: int
    detections_json: list[DetectionItemSchema]
    inference_time_ms: Optional[int] = None
    image_path: Optional[str] = None
    annotated_image_path: Optional[str] = None
    video_frame_index: Optional[int] = None
    device_id: Optional[str] = None
    created_by: Optional[str] = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True, protected_namespaces=())

File: GameAssistant/web/frontend/src/api/inference.ts
import request from './request'

export interface DetectionItem {
  cls: string
  x: number
  y: number
  w: number
  h: number
  conf: number
  source: string
}

export interface InferenceResult {
  id: string
  name?: string
  source_type: string
  source_file?: string
  model_id?: string
  inference_mode: string
  confidence_threshold: number
  total_detections: number
  detections_json: DetectionItem[]
  inference_time_ms?: number
  image_path?: string
  annotated_image_path?: string
  video_frame_index?: number
  device_id?: string
  created_by?: string
  created_at: string
}

export interface InferenceStats {
  total: number
  device: number
  image: number
  video: number
}

export const inferenceApi = {
  async getList(params?: {
    page?: number
    page_size?: number
    source_type?: string
    model_id?: string
  }): Promise<InferenceResult[]> {
    const q = new URLSearchParams()
    if (params?.page) q.set('page', String(params.page))
    if (params?.page_size) q.set('page_size', String(params.page_size))
    if (params?.source_type) q.set('source_type', params.source_type)
    if (params?.model_id) q.set('model_id', params.model_id)
    return request.get(`/inference/${q.toString() ? `?${q}` : ''}`)
  },

  async create(data: Partial<InferenceResult>): Promise<InferenceResult> {
    return request.post('/inference/', data)
  },

  async delete(id: string): Promise<void> {
    return request.delete(`/inference/${id}`)
  },

  async getStats(): Promise<InferenceStats> {
    return request.get('/inference/stats/summary')
  },
}

export default inferenceApi
